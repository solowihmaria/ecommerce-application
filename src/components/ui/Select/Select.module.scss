@use '../../../assets/styles/constants.module' as c;

.selectContainer {
    display: flex;
    flex-direction: column;
    gap: 0.25rem;
    width: 100%;
    max-width: 16rem;
}

.selectWrapper {
    position: relative;
    width: 100%;
}

.select {
    box-sizing: border-box;
    width: 100%;
    padding: 0 0.5rem;
    border-color: transparent;
    border-width: 3px;
    border-radius: 1rem;
    font-family: inherit;
    font-size: 1rem;
    background-color: c.$color-primary-light;
    transition: border-color 0.2s ease;

    &:focus {
        border-color: c.$color-accent-primary;
        outline: none;
    }

    &:disabled {
        cursor: not-allowed;

        // background-color: c.$color-gray-100;
        opacity: 0.7;
    }

    &.error {
        border-color: c.$color-error;
    }
}

.errorMessage {
    display: block;
    width: 100%;
    min-height: 1.25rem;
    border-color: c.$color-error;
    font-size: 0.7rem;
    overflow-wrap: break-word;
    white-space: normal;
}
