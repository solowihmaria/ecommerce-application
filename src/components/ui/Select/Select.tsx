import styles from './Select.module.scss';
import clsx from 'clsx';
import type { SelectProps } from './Select.types';

export const Select = ({
    name,
    className = '',
    required,
    disabled,
    ...props
}: SelectProps) => {
    return (
        <div className={styles.selectWrapper}>
            <select
                data-testid="test-id-select-field"
                name={name}
                className={clsx(
                    styles.select,
                    className,
                    disabled && styles.disabled,
                    required && styles.required
                )}
                disabled={disabled}
                required={required}
                {...props}
            ></select>

            {required && <span className={styles.requiredMarker}>*</span>}
        </div>
    );
};
