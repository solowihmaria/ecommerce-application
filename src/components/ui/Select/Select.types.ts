import type { ReactNode } from 'react';

export interface SelectProps {
    name: string;
    children: ReactNode;
    className?: string;
    required?: boolean;
    disabled?: boolean;
}
