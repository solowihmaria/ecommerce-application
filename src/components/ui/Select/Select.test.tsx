import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Select } from './Select';

describe('<Select/> component', () => {
    test('Render Select component', () => {
        render(<Select name="selectTest" />);
        const select = screen.getByTestId('test-id-select-field');
        expect(select).toBeInTheDocument();
        expect(select).toMatchSnapshot();
    });
});
