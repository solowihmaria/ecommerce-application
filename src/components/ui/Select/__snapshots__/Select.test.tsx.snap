// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select/> component Render Select component 1`] = `
<select
  class="select"
  data-testid="test-id-select-field"
  name="selectTest"
/>
`;
